src/metabase/util/yaml.clj

```diff
@@ -45,3 +45,13 @@
                                    u/pprint-to-str)
                            e)))
        (throw e)))))
+
+(defn load-dir
+  "Load and parse all YAMLs in `dir`. Optionally pass each resulting data structure through `constructor-fn`."
+  ([dir] (load-dir dir identity))
+  ([dir constructor]
+   (with-resource [dir dir]
+     (with-open [ds (Files/newDirectoryStream dir)]
+       (->> ds
+            (filter (comp #(str/ends-with? % ".yaml") str/lower-case (memfn ^Path getFileName)))
+            (mapv (partial load constructor)))))))
```

Comment:
I don't know if this is already the case (and, if not, not really 100% relevant to this PR), but it would be really cool if the logic for generating the YAML was 100% separate from writing to a file. The YAML generation logic would output a string and some other function would handle writing the string to a file. That way we could easily test the output without having to worry about reading it from files. And debugging in the REPL would be easier too.

If you really wanted to get Lispy you could just have the YAML generation code write everything to standard out with printlns and the like and then wrap the call in `with-out-str` if you wanted a string, or otherwise rebind `*out*` as needed if you wanted to do something wacky like stream the YAML as it's generated. That would make debugging even easier than always generating a string

Reply:
:whatcamsaid: